### ISSUE TITLE
OpenShift resources in Kubernetes YAML

### ISSUE THREAD

**Issue by Ladicek** [dekorateio/dekorate#367]
To reproduce:

```bash
mvn clean install -DskipTests -DskipITs
# Maven invoker plugin isn't disabled, so invoker-based tests are executed
vim frameworks/thorntail/target/it/annotationless-yml/target/classes/META-INF/dekorate/kubernetes.yml
# observe OpenShift resources, e.g. image streams
```

(I'd like to point out that editing the invoker-based integration tests in an IDE is unnecessarily hard. Surely there's a better way to set up integration tests. At the moment, I can't think of a reason why plain old Maven modules wouldn't work. They could be put, as submodules, into a separate `integration-tests` module, so they would have a common parent POM which could set up the necessary infrastructure. I understand that this is not entirely desirable, as it breaks test isolation, but it's also pretty common.)

**Comment by iocanel** [comment]
@Ladicek: we have decided to move away from `maven-invoker-plugin` its just a matter of someone doing it :-D

**Comment by Ladicek** [comment]
Great to hear!

**Comment by iocanel** [comment]
We have two things here:

1. dekorate-thorntail depends on openshift-annotations (it should be optional).
2. dekorate will add s2i resources everywhere if s2i-annotations is found in the classpath.

Point 1. is easily fixable.
Point 2 needs more thought. The original intention was to be able to mix and match and do things like `knative` with `s2i`. Unfortunately, this doesn't play well unless we push to a docker registry. Now, I am not sure whats the best way to handle it. Add s2i resources to their own file? Limit them just to openshfit.yml/json?

Maybe, I'll limit them to openshift.yml at least for now as its the less disruptive change.

**Comment by Ladicek** [comment]
Hmm, I'm not sure why I added a dependency on `openshift-annotations` to `dekorate-thorntail`, I would have sweared that I've seen it somewhere else :-)

Agree with limiting the S2I resources to `openshift.yml`.

**Comment by iocanel** [comment]
Fixed as of 91461749c6708c72b7a884b0f430d927b7f61791 and released in 0.9.3
